scripts/ci/issue-report/generateReport.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { Octokit } = require("@octokit/rest");
+const nodemailer = require('nodemailer');
+
+function sendReport(title, header, issues) {
+    if (!issues || issues.length == 0) {
+        return;
+    }
+    let report = header + "\n\n"
+    for (const issue of issues) {
+        report += `${issue.url}: ${issue.title}\n`;
+    }
+      
+      nodemailer.createTransport({
+        service: process.env['ISSUE_REPORT_SENDER_EMAIL_SERVICE'], // e.g. "gmail"
+        auth: {
+          user: process.env['ISSUE_REPORT_SENDER_EMAIL_ADDRESS'],
+          pass: process.env['ISSUE_REPORT_SENDER_EMAIL_PASSWORD']
+        }
+      }).sendMail({
+        from: process.env['ISSUE_REPORT_SENDER_EMAIL_ADDRESS'],
+        to: process.env['ISSUE_REPORT_RECIPIENT_EMAIL_ADDRESS'],
+        subject: title,
+        text: report
+      }, function(error, info){
+        if (error) {
+          console.log(error);
+        } else {
+          console.log('Email sent: ' + info.response);
```

Comment:
Presumably we don't want to get a notification when there is a success. So I'm guessing both the above and this are both only visible when someone opts in to debugging. Ideally we would get an explicit notification of failure somehow (probably not via an emailed report, if we failed to email a report :-).